File: services/product-service/app/src/main/java/io/micro_spring_boot/product_service/app/util/DataLoader.java
package io.micro_spring_boot.product_service.app.util;

import io.micro_spring_boot.product_service.domain.aggregates.product.Product;
import io.micro_spring_boot.product_service.domain.aggregates.product.ProductType;
import io.micro_spring_boot.product_service.infra.repositories.ProductRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class DataLoader implements CommandLineRunner {
    private final ProductRepository productRepository;

    @Override
    public void run(String... args) {
        if (productRepository.count() < 1) {
            Product[] products = {
                new Product(
                    "iphone_13",
                    "iPhone 13",
                    "iPhone 13",
                    1000d,
                    ProductType.SINGLE
                ),
                new Product(
                    "iphone_13_pro",
                    "iPhone 13 Pro",
                    "iPhone 13 Pro",
                    1200d,
                    ProductType.SINGLE
                ),
                new Product(
                    "iphone_13_pro_max",
                    "iPhone 13 Pro Max",
                    "iPhone 13 Pro Max",
                    1300d,
                    ProductType.SINGLE
                ),
                new Product(
                    "iphone_13_mini",
                    "iPhone 13 Mini",
                    "iPhone 13 Mini",
                    900d,
                    ProductType.SINGLE
                )
            };

            productRepository.saveAll(List.of(products));
        }
    }
}

File: services/product-service/app/src/main/kotlin/com/microspringboot/productservice/app/utils/DataLoader.kt
package com.microspringboot.productservice.app.utils

import com.microspringboot.productservice.domain.aggregates.products.Product
import com.microspringboot.productservice.domain.aggregates.products.ProductType
import com.microspringboot.productservice.infra.repositories.ProductRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
@Suppress("MagicNumber")
class DataLoader (
    private val productRepository: ProductRepository
) : CommandLineRunner {
    override fun run(vararg args: String) {
        if (productRepository.count() < 1) {
            val products = listOf(
                Product(
                    "iphone_13",
                    "iPhone 13",
                    "iPhone 13",
                    1000.0,
                    ProductType.SINGLE
                ),
                Product(
                    "iphone_13_pro",
                    "iPhone 13 Pro",
                    "iPhone 13 Pro",
                    1200.0,
                    ProductType.SINGLE
                ),
                Product(
                    "iphone_13_pro_max",
                    "iPhone 13 Pro Max",
                    "iPhone 13 Pro Max",
                    1300.0,
                    ProductType.SINGLE
                ),
                Product(
                    "iphone_13_mini",
                    "iPhone 13 Mini",
                    "iPhone 13 Mini",
                    900.0,
                    ProductType.SINGLE
                )
            )
            productRepository.saveAll(products)
        }
    }
}
